src/fontra/views/editor/editor.js

```diff
@@ -1445,13 +1477,29 @@ export class EditorController {
       transformation: transformation,
       location: location,
     };
-
-    await this.sceneController.editInstanceAndRecordChanges((instance) => {
-      const newComponentIndex = instance.components.length;
-      instance.components.push(newComponent);
-      this.sceneController.selection = new Set([`component/${newComponentIndex}`]);
-      return "Add Component";
-    });
+    if (addToAllSources) {
+      await this.sceneController.editGlyphAndRecordChanges((glyph) => {
+        for (const layerName in glyph.layers) {
```

Comment:
There is not necessarily a 1-to-1 relationship between sources and layers. We only want to add the component to *source* layers. To make it even more complicated, multiple sources _may_ reference the same layer. 

What I think you should do is something like this:
```js
    const layerNames = new Set()
    for (const source of glyph.sources) {
      layerNames.add(source.layerName)
    }
    for (const layerName in layerNames) {
      /* etc */
    }
```